jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml

```diff
@@ -43,17 +44,31 @@
 - project:
     name: kubernetes-builds
     build:
-        - 'build':
-            branch: 'master'
-            timeout: 50
-        - 'build-1.0':
-            branch: 'release-1.0'
-            timeout: 30
-        - 'build-1.1':
-            branch: 'release-1.1'
-            timeout: 30
-        - 'build-1.2':
-            branch: 'release-1.2'
-            timeout: 30
+      - 'build':
+          url: 'https://github.com/kubernetes/kubernetes'
+          branch: 'master'
+          timeout: 50
+      - 'build-1.0':
+          url: 'https://github.com/kubernetes/kubernetes'
+          branch: 'release-1.0'
+          timeout: 30
+      - 'build-1.1':
+          url: 'https://github.com/kubernetes/kubernetes'
+          branch: 'release-1.1'
+          timeout: 30
+      - 'build-1.2':
+          url: 'https://github.com/kubernetes/kubernetes'
+          branch: 'release-1.2'
+          timeout: 30
+
+      - 'federation-build':
+          branch: 'master'
+          url: 'https://github.com/kubernetes/kubernetes'
+          timeout: 50
+          job-env: |
+            export PROJECT="k8s-jkns-e2e-gce-federation"
```

Comment:
We have added a [step](https://github.com/kubernetes/kubernetes/pull/25125/files#diff-224de03926628a1859f1c1a768195e7aR51) to `build/push-ci-build.sh` which will push federated container images to a docker repo following a successful build.

In the `KUBERNETES_PROVIDER=GCE` case, `gcloud docker push` is used to automatically deal with credentials. I would think that `gcloud` at this point needs to know which project context it's operating in via `PROJECT`.